data/changelog/posts/dune-developer-preview/2025-06-10-migrating-ocaml-org.md

```diff
@@ -0,0 +1,164 @@
+
+**Step-by-step workspace configuration for OCaml.org:**
+
+1. **Pin opam-repository to the same specific commit** in the `dune-workspace` files:
+   ```lisp
+   ; Both dune-workspace and playground/dune-workspace use:
+   (repository
+    (name pinned_opam_repository)
+    (url git+https://github.com/ocaml/opam-repository#584630e7a7e27e3cf56158696a3fe94623a0cf4f))
+
+   (lock_dir
+    (path dune.lock)
+    (repositories 
+     pinned_opam_repository 
+      ))
+   ```
+
+2. **Update the pinned repository commit** in both files during coordinated upgrades:
+   ```bash
+   # Update both dune-workspace files with the new commit hash
+
+   # Then regenerate both lock files
+   dune pkg lock
+   cd playground && dune pkg lock --root .
+   ```
+
+### Pinning Individual Dependencies
+
+The playground has additional complexity because it requires specific versions of packages that aren't available in the standard `opam-repository` or need patches for compatibility with Dune package management. These dependencies are handled through individual pins in the playground's `dune-workspace`.
+
+**Pin a dependency**:
+
+```
+...
+
+(pin
+(name merlin-js)
+ (url "git+https://github.com/voodoos/merlin-js#3a8c83e03d629228b8a8394ecafc04523b0ab93f")
+ (package
+  (name merlin-js)))
+
+...
+
+(lock_dir
+  (repositories pinned_opam_repository)
+  (pins esbuild code-mirror merlin-js js-top-worker ocamlbuild ocamlfind)
+   (version_preference newest))
+```
+
+
+### Using the Opam-repository Overlays for OCamlBuild and OCamlFind
+
+When we pinned `opam-repository` to a fixed git commit in the `dune-workspace` files, we encountered errors relating to `ocamlbuild` and `ocamlfind`. This is to be expected, as the most recent released versions of `ocamlbuild` and `ocamlfind` are not compatible with Dune package management yet. This is why a repository [`opam-overlays`](https://github.com/ocaml-dune/opam-overlays) has been created, where packages are republished with fixes relating to Dune package management.
+
+So, we added the overlay repository to the `dune-workspace` files and listed both repositories within the `lock_dir` Dune stanza:
+
+```
+...
+
+(repository (name pinned_overlay_repository) (url git+https://github.com/ocaml-dune/opam-overlays#2a9543286ff0e0656058fee5c0da7abc16b8717d))
+
+(lock_dir
+  (repositories pinned_opam_repository pinned_overlay_repository))
+```
+
+This way, the patched dependencies `ocamlbuild` and `ocamlfind` are picked up by Dune's solver when creating the `dune.lock` directory.
+
+### Dependency Compatibility
+
+The migration uncovered compatibility issues with upstream dependencies. Specifically, the `merlin-js` dependency, which hasn't been upstreamed to the main Merlin project, appears to have compatibility issues with Dune package management. This highlights an important consideration for the broader ecosystem: a few packages may need updates to work seamlessly with the new package management features.
+
```

Comment:
The issue with `merlin-js` was that it used symlinks in its example programs. Dune package management doesn't support this at the moment, so we removed this to enable building it with Dune package management. This issue is similar to the issue with `ocamlbuild` and `ocamlfind`; both packages have patched versions removing symlinks for dune package management.

Reply:
Thanks!